runtime/indent/testdir/vim.in

```diff
@@ -956,6 +956,7 @@ FIVE(1 + 2 + 2), SIX(36 / 3 / 2), SEVEN(7), EIGHT(2 * 2 * 2),
 NINE(3 + 3 + 3)
 const value: number
 def new(this.value)
+return this.value
```

Comment:
```suggestion
def new(value: number)
this.value = value
```
This is a constructor definition (`:help E1365`).

Reply:
@zzzyxwvut `def new(this.value)` is a valid constructor syntax. However, neither return nor assignment is required.
See `:h E1390`.

Reply:
The [suggested test](https://github.com/vim/vim/pull/16293/commits/8737bb883d9486922caccb653ca98494bb28d196) had made use of that syntax.  However,  
with a [newer commit](https://github.com/vim/vim/pull/16293/commits/f0549fe57d693864b2a93a94dce9cf53ca7329e5), additional indentation was introduced  
for testing with a `return this.value` statement; and `return`  
_with a value_ is malformed syntax for constructors.  In order  
to keep that indentation and use valid syntax, the above  
suggestion was offered.

Otherwise, constructors equally support shorthand assignment  
and/or regular parameters, as well as `return`s _with no value_.

For example,

```vim
vim9script

enum Test
    INSTANCE(1260, "twelve hundred and threescore", true)

    const number: number
    const string: string

    def new(this.number, string: string, _)
        if strlen(string) < 24
            this.string = string
        else
            return
        endif
    enddef
endenum
```